docs/features/deconstruction.md

```diff
@@ -135,59 +135,56 @@ evaluate Deconstruct passing the references directly in
-Also, the `Deconstruct` method must be an instance method or an extension (but not a static method).
+Also, the `Deconstruct` method must be an instance method or an extension (but not a static method). It also must return `void`.
 
 
 ###Deconstruction-declaration (deconstruction into new variables):
 
+The *deconstruction-declaration* is also represented with an *assignment*, but the left-hand-side is either a *declaration-expression* or a tuple expression containing *declaration-expressions*.
 *Deconstruction-declarations* can be thought of as two steps: (1) declaring new locals, and (2) applying a *deconstruction-assignment* into those locals.
-The syntax for deconstruction is more involved and multiple forms are allowed. The simplest case is `(int x, string y)`. Variants include nested declarations like `(int x, (string y, long z))` (which declares 3 locals) and implicitly typed declarations like `(var x, var y)`. The latter can also be written using the shorthand `var (x, y)`.
-`var` is the only case where such shorthand is allowed (you `int (x, y)` is not legal).
+The declaration of new locals by the left-hand-side of the assignment can take multiple forms. The simplest case is `(int x, string y)`, that is a tuple expression containing declaration expressions. Variants include nested declarations like `(int x, (string y, long z))` (which declares 3 locals) and implicitly typed declarations like `(var x, var y)`. The latter can also be written using the shorthand `var (x, y)`, which is a declaration expression with parenthesized designations.
+`var` is the only case where such shorthand is allowed (so `int (x, y)` is not legal).
 
 As in the case of *deconstruction-declarations*, tuple expressions on the right-hand-side have their type inferred from the left-hand-side. With *deconstruction-declaration*, this is also the case, except that any type is `var` in the left-hand-side, then the natural type of the element in the right-hand-side is used.
 For example, in `(string x, byte y, var z) = (null, 1, 2);`, `null` has type `string`, the literal `1` has type `byte` (inferred from `y`) and the literal `2` has type `int` (its natural type).
 
+In C#7.0, *deconstruction-declarations* are only allowed as top-level statements. They do not allow mixing of declaration expressions and assignable expressions (such as `(existing, var declared) = (1, 2)`).
+
 ###Grammar changes
 
 ```ANTLR
-deconstruction_declaration_statement // new
-    : modifiers variable_component_assignment ';'
-    ;
-    
-variable_component_assignment // new
-    : variable_component = expression
-    ;
-    
-variable_component // new
-    : type variable_designation
-    | '(' variable_component (',' variable_component)+ ')'
-    | '*' // wildcard (future)
-    ;
+expression
+	: ... // existing
+	| declaration_expression // new
```

Comment:
You should probably add some text explaining that a declaration expression is only permitted in specific designated syntactic positions.